lib/rubocop/cop/betterment/unscoped_find.rb

```diff
@@ -49,11 +49,18 @@ def on_send(node)
                 static_method_name(node.method_name)
             ) && !@unauthenticated_models.include?(Utils::Parser.get_root_token(node))
 
-          add_offense(node) if find_param_arg(arg_nodes)
+          add_offense(node) if find_param_arg(arg_nodes) || graphql_namespace?(node)
         end
 
         private
 
+        def graphql_namespace?(node)
+          node
+            .ancestors
+            .select { |n| n.class_type? || n.module_type? }
+            .any? { |n| n.identifier.to_s.downcase.include?("graphql") }
```

Comment:
__question:__ Just curious if we should match the string if it contains `"graphql"` string *anywhere* even if it's a subset of some larger string?

What if instead we used:

```ruby
n.identifier.to_s.match?(/\bGraphQL\b/)
```

If we want to match `GraphQL` and `Graphql` or similar, then we could add the `/i` regexp modifier to make it case insensitive.

Reply:
Mm interesting, so I had intentionally written it to be at least flexible enough to match `GraphQL` and `Graphql`, thinking that I could easily imagine someone making the choice to downcase the `QL`, and thinking that that decision didn't feel super relevant to me, as to whether this cop should still apply.

Down to hear any other opinions on the question! Either way, I think the regex is a nicer touch (particularly with the word boundary) so I can switch to the case insensitive version and then see what others think?

Reply:
I should note that I think the odds of another string containing the subset `graphql`, but it being something else, is pretty low. It's more as a matter of practice that I try to match strings in a way that would be correct in more instances.

I *have* seen production bugs caused by using `String#include?` where a stricter match wouldn't have.